docs(nuxt): update 7.state-management.md (#28053)
nuxt/nuxt: docs/1.getting-started/7.state-management.md

## Best Practices

::warning
Never define `const state = ref()` outside of `<script setup>` or `setup()` function.

> For example, doing `export myState = ref({})` would result in state shared across requests on the server.<br>

Reviewer: ```suggestion
For example, doing `export myState = ref({})` would result in state shared across requests on the server and can lead to memory leaks.
```

> Such state will be shared across all users visiting your website and can lead to memory leaks!

Reviewer: ```suggestions
```

## Best Practices

::warning

> Never define `const state = ref()` outside of `<script setup>` or `setup()` function.

Reviewer: ```suggestion
Never define `const state = ref()` outside of `<script setup>` or `setup()` function.<br>
```
